### ISSUE TITLE
[ENH] Multivariate decoding for FREM

### ISSUE THREAD

**Issue by Aceticia** [nilearn/nilearn#4177]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe your proposed enhancement in detail.

Nilearn provides a very robust solution to univariate decoding with tools such as searchlight and spacenet. But they are currently limited to one variable at a time. I would like decoding methods here to support multiple targets. According to my possibly naive understanding of the repo, changes to the code should be minimal.

### Benefits to the change

With more complex research topics emerge such as deep neural network based research, it would greatly benefit the popularity of nilearn among DNN and machine learning researchers (me included). Right now this is doable, but not straightforward. A workaround I have is that I run individual variables, and combine them to get an approximation of the real accuracy map.

**Comment by Aceticia** [comment]
I'm down to help implement this if that's something people are interested in having.

**Comment by bthirion** [comment]
Thx for posting. 
Can you specify which model you would like ? 
The idea is that you can easily call all sklearn compatible estimators using  the approach in https://nilearn.github.io/stable/auto_examples/02_decoding/plot_simulated_data.html
Best,

**Comment by Aceticia** [comment]
Hi, thanks for the quick response. I think this is trivially doable with standard models such as linear regression or even NN. But I don't think more advanced methods such as spacenet or searchlight would work directly with multiple targets.

**Comment by bthirion** [comment]
Thx for the clarification.
I don't think that we want to enhance SpaceNet, which is technically challenging and definitely not great regarding cost/benefit tradeoffs. Instead, it makes sense to enhance FreM. Need to check how good it is regarding multi-valued outputs.

**Comment by Aceticia** [comment]
I'm not very familiar with FREM and I might not be helpful in implementing this. I do agree though modifying spacenet would be a major hassle.

**Comment by bthirion** [comment]
Question: is there any dataset on which we could run a multi-task model ? It makes little sense to support a feature if we cannot check that it does a reasonable job.

**Comment by Remi-Gau** [comment]
@bthirion 

I think this one does have several tasks:

https://nilearn.github.io/dev/modules/generated/nilearn.datasets.fetch_ds000030_urls.html#nilearn.datasets.fetch_ds000030_urls

used in here:

https://nilearn.github.io/dev/auto_examples/04_glm_first_level/plot_bids_features.html

dataset descriptor: https://nilearn.github.io/dev/auto_examples/04_glm_first_level/plot_bids_features.html

**Comment by bthirion** [comment]
Thx but we want a decoding setting.

**Comment by bthirion** [comment]
I'm considering the Miyawaki dataset.
